ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/RegexpSplit.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.string;
+
+import com.google.common.base.Splitter;
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@UdfDescription(name = "regexp_split",
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "Splits a string into an array of substrings based on a regexp. "
+        + "If the regexp is found at the beginning of the string, end of the string, or there "
+        + "are contiguous matches in the string, then empty strings are added to the array. "
+        + "If the regexp is not found, then the original string is returned as the only "
+        + "element in the array. If the regexp is empty, then all characters in the string are "
+        + "split.")
```

Comment:
Add the bit about empty adding empty elements from the `syntax-reference.md` in here to?

Reply:
> If the regexp is found at the beginning of the string, end of the string, or there 
> are contiguous matches in the string, then empty strings are added to the array.

It's there in the beginning.